src/services/fetchers/BinancePriceFetcher.ts

```diff
@@ -46,48 +48,63 @@ export default class BinancePriceFetcher implements FeedFetcherInterface {
       timeoutErrorMessage: `Timeout exceeded: ${sourceUrl}`,
     });
 
-    if (response.status !== 200) {
-      throw new Error(response.data);
-    }
+    const timestamp = this.timeService.apply();
+    const parsed = this.parseResponse(response);
+    await this.savePrices(timestamp, parsed);
 
-    if (response.data.Response === 'Error') {
-      throw new Error(response.data.Message);
-    }
+    const prices = await this.binanceDataRepository.getPrices(inputs, timestamp);
 
-    const fetcherResult = {
-      prices: this.resolveFeeds(inputs, response.data as BinanceResponse),
-      timestamp: this.timeService.apply(),
+    const fetcherResults: FetcherResult = {
+      prices: prices.map((price, ix) => (price && inputs[ix].inverse ? 1.0 / price : price)),
```

Comment:
shouldn't the operator be an `or` (`||`) ? It's for me a bit tricky to understand the result of this operation.

Reply:
I changed to more readable, however result is exactly the same.